pkg/chartmuseum/server/multitenant/handlers.go

```diff
@@ -33,6 +33,7 @@ var (
 	objectSavedResponse   = gin.H{"saved": true}
 	objectDeletedResponse = gin.H{"deleted": true}
 	healthCheckResponse   = gin.H{"healthy": true}
+	versionResponse       = gin.H{"version": "0.2.0"}
```

Comment:
Any suggestions about where I can get this info...dynamically?

I just learned that you cannot import the main package elsewhere so I can't import.
https://github.com/helm/chartmuseum/blob/b5bb97f67466f636f8c48f7dcf09b766de2516cd/cmd/chartmuseum/main.go#L39

I explored `"github.com/urfave/cli"`'s `App.Version` but I don't have a ref to `App` instance either, and there doesn't seem to be an easy way to get it.

Or is it fine to have it static?

Reply:
The `Version` is defined in the Makefile , add passed through `main.Version` (as well as the `Reversion`). So it's OK if U can pass the `main.Version` through the server options (and then multiTenantServer option) ? Thus U can implement the version as the "Dynamic" version .

Reply:
But a little tricky XD

Reply:
@scbizu I tried, but it doesn't work and I've no idea what is going wrong. lol

```diff
diff --git a/cmd/chartmuseum/main.go b/cmd/chartmuseum/main.go
index b91f9a2..58b6ae6 100644
--- a/cmd/chartmuseum/main.go
+++ b/cmd/chartmuseum/main.go
@@ -62,7 +62,10 @@ func cliHandler(c *cli.Context) {
 	backend := backendFromConfig(conf)
 	store := storeFromConfig(conf)
 
+	fmt.Println("DEBUG, main version is", Version) // version: 0.2.0
+
 	options := chartmuseum.ServerOptions{
+		Version:                Version,
 		StorageBackend:         backend,
 		ExternalCacheStore:     store,
 		TimestampTolerance:     conf.GetDuration("storage.timestamptolerance"),
diff --git a/pkg/chartmuseum/server.go b/pkg/chartmuseum/server.go
index 3017f43..527e567 100644
--- a/pkg/chartmuseum/server.go
+++ b/pkg/chartmuseum/server.go
@@ -17,6 +17,7 @@ limitations under the License.
 package chartmuseum
 
 import (
+	"fmt"
 	"strings"
 	"time"
 
@@ -70,6 +71,7 @@ type (
 		// More refers to : https://github.com/helm/chartmuseum/issues/320
 		EnforceSemver2 bool
 		Host           string
+		Version        string
 	}
 
 	// Server is a generic interface for web servers
@@ -118,6 +120,8 @@ func NewServer(options ServerOptions) (Server, error) {
 		Host:              options.Host,
 	})
 
+	fmt.Println("DEBUG, options version is", options.Version) // version: 0.2.0
+
 	server, err := mt.NewMultiTenantServer(mt.MultiTenantServerOptions{
 		Logger:                 logger,
 		Router:                 router,
@@ -136,7 +140,10 @@ func NewServer(options ServerOptions) (Server, error) {
 		AllowOverwrite:         options.AllowOverwrite,
 		AllowForceOverwrite:    options.AllowForceOverwrite,
 		EnforceSemver2:         options.EnforceSemver2,
+		Version:                options.Version,
 	})
 
+	fmt.Println("DEBUG, server version is", server.Version) // version: ""
+
 	return server, err
 }
diff --git a/pkg/chartmuseum/server/multitenant/handlers.go b/pkg/chartmuseum/server/multitenant/handlers.go
index 960af94..cf57bff 100644
--- a/pkg/chartmuseum/server/multitenant/handlers.go
+++ b/pkg/chartmuseum/server/multitenant/handlers.go
@@ -33,7 +33,6 @@ var (
 	objectSavedResponse   = gin.H{"saved": true}
 	objectDeletedResponse = gin.H{"deleted": true}
 	healthCheckResponse   = gin.H{"healthy": true}
-	versionResponse       = gin.H{"version": "0.2.0"}
 	welcomePageHTML       = []byte(`<!DOCTYPE html>
 <html>
 <head>
@@ -81,6 +80,7 @@ func (server *MultiTenantServer) getWelcomePageHandler(c *gin.Context) {
 }
 
 func (server *MultiTenantServer) getInfoHandler(c *gin.Context) {
+	versionResponse := gin.H{"version": server.Version} // version: "" :(
 	c.JSON(200, versionResponse)
 }
 
diff --git a/pkg/chartmuseum/server/multitenant/server.go b/pkg/chartmuseum/server/multitenant/server.go
index 741ba95..b255ff1 100644
--- a/pkg/chartmuseum/server/multitenant/server.go
+++ b/pkg/chartmuseum/server/multitenant/server.go
@@ -62,6 +62,7 @@ type (
 		ChartURL               string
 		ChartPostFormFieldName string
 		ProvPostFormFieldName  string
+		Version                string
 		Limiter                chan struct{}
 		Tenants                map[string]*tenantInternals
 		TenantCacheKeyLock     *sync.Mutex
@@ -77,6 +78,7 @@ type (
 		ChartURL               string
 		ChartPostFormFieldName string
 		ProvPostFormFieldName  string
+		Version                string
 		MaxStorageObjects      int
 		IndexLimit             int
 		GenIndex               bool

```

Reply:
Nevermind xD Just tired

https://github.com/helm/chartmuseum/pull/358/commits/ad023a1b1b0b785e9713b315c2d8c725e123b560

(forgot to add `Version` to NewMultiTenantServer)